daemon/algod/api/server/v2/dryrun_test.go

```diff
@@ -1091,3 +1091,109 @@ int 1`)
 		logResponse(t, &response)
 	}
 }
+
+func TestDryrunLogs(t *testing.T) {
+	t.Parallel()
+
+	ops, err := logic.AssembleString(`
+#pragma version 5
+int 1
+byte "a"
+log
+byte "b"
+log
+byte "c"
+log
+`)
+
+	require.NoError(t, err)
+	approval := ops.Program
+	ops, err = logic.AssembleString("int 1")
+	clst := ops.Program
+	ops, err = logic.AssembleString("#pragma version 5 \nint 1")
+	approv := ops.Program
+	require.NoError(t, err)
+
+	var appIdx basics.AppIndex = 1
+	creator := randomAddress()
+	sender := randomAddress()
+	dr := DryrunRequest{
+		Txns: []transactions.SignedTxn{
+			{
+				Txn: transactions.Transaction{
+					Header: transactions.Header{Sender: sender},
+					Type:   protocol.ApplicationCallTx,
+					ApplicationCallTxnFields: transactions.ApplicationCallTxnFields{
+						ApplicationID: appIdx,
+						OnCompletion:  transactions.OptInOC,
+					},
+				},
+			},
+			{
+				Txn: transactions.Transaction{
+					Header: transactions.Header{Sender: sender},
+					Type:   protocol.ApplicationCallTx,
+					ApplicationCallTxnFields: transactions.ApplicationCallTxnFields{
+						ApplicationID: appIdx + 1,
+						OnCompletion:  transactions.OptInOC,
+					},
+				},
+			},
+		},
+		Apps: []generated.Application{
+			{
+				Id: uint64(appIdx),
+				Params: generated.ApplicationParams{
+					Creator:           creator.String(),
+					ApprovalProgram:   approval,
+					ClearStateProgram: clst,
+					LocalStateSchema:  &generated.ApplicationStateSchema{NumByteSlice: 1},
+				},
+			},
+			{
+				Id: uint64(appIdx + 1),
+				Params: generated.ApplicationParams{
+					Creator:           creator.String(),
+					ApprovalProgram:   approv,
+					ClearStateProgram: clst,
+					LocalStateSchema:  &generated.ApplicationStateSchema{NumByteSlice: 1},
+				},
+			},
+		},
+		Accounts: []generated.Account{
+			{
+				Address: sender.String(),
+				Status:  "Online",
+				Amount:  10000000,
+			},
+		},
+	}
+	dr.ProtocolVersion = string(dryrunProtoVersion)
+
+	var response generated.DryrunResponse
+	doDryrunRequest(&dr, &response)
+	require.NoError(t, err)
+	checkAppCallPass(t, &response)
+	if t.Failed() {
+		logResponse(t, &response)
+	}
+	logs := *response.Txns[0].Logs
+	assert.Equal(t, 3, len(logs))
+	for i, m := range logs {
+		if i == 0 {
+			assert.Equal(t, base64.StdEncoding.EncodeToString([]byte("a")), m.Value)
+		} else if i == 1 {
+			assert.Equal(t, base64.StdEncoding.EncodeToString([]byte("b")), m.Value)
+		} else if i == 2 {
+			assert.Equal(t, base64.StdEncoding.EncodeToString([]byte("c")), m.Value)
+		}
+
+	}
```

Comment:
Calculating values looks more maintainable, check below:
```suggestion
	for i, m := range logs {
		assert.Equal(t, base64.StdEncoding.EncodeToString([]byte(string(rune('a' + i)))), m.Value)
	}
```

If logs are not sorted, use sort.Slice. If logs is map (not sure) then flatten it:
```golang
	type flog struct {
		idx int,
		li LogItem
	}
	for k, v := range logs {
		flatlog = append(flatlog, flog{i, v})
	}
	sort.Slice(flatlog, func(i, j int) {return flatlog[i].idx < flatlog[j].idx})
```

Reply:
The order of logs matter, so they should not be sorted before comparison

Reply:
well ok, then `assert.Equal(t, base64.StdEncoding.EncodeToString([]byte(string(rune('a' + i)))), m.Value)` will work just fine